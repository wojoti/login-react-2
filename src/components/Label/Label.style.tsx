import styled from "styled-components";
type Props = {
  color?: string;
};
export const Wrapper = styled.span<Props>`
  color: ${(props) => props.color || "inherit"};
  font-size: 14px;
`;
