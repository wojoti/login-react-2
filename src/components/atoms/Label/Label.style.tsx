import styled from 'styled-components'
interface Props {
  color?: string
}
export const StyledLabel = styled.span<Props>`
  color: ${(props) => props.color || 'inherit'};
  font-size: 14px;
`
