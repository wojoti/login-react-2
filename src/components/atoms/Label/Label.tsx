import StyledLabel from "./Label.style";

export interface LabelProps {
  children?: React.ReactNode;
  color?: string;
  testId?: string;
}

function Label({ children, color, testId }: LabelProps) {
  return (
    <StyledLabel color={color} data-testid={testId}>
      {children}
    </StyledLabel>
  );
}

export default Label;
