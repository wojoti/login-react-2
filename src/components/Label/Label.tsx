import React, { ReactNode } from "react";
import { Wrapper } from "./Label.style";

type Props = {
  children: ReactNode;
};

const Label = (props: Props) => {
  return <Wrapper>{props.children}</Wrapper>;
};

export default Label;
