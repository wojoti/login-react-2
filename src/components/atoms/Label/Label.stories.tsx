/* eslint-disable react/function-component-definition */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-extraneous-dependencies */
import { ComponentMeta, ComponentStory } from "@storybook/react";

import Label from "./Label";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Atoms/Label",
  component: Label,
  decorators: [
    (Story) => (
      <div style={{ display: "flex", justifyContent: "center" }}>
        <Story />
      </div>
    ),
  ],
} as ComponentMeta<typeof Label>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Label> = (args) => <Label {...args} />;

export const Email = Template.bind({});
Email.args = { children: "Email" };

export const Password = Template.bind({});
Password.args = { children: "Password" };

export const RememberMe = Template.bind({});
RememberMe.args = { children: "Remember Me?" };
