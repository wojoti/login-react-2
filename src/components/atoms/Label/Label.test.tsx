import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Label, { LabelProps } from "./Label";

const props: LabelProps = {
  testId: "test-label-id",
};

const modifiedProps: LabelProps = {
  ...props,
  color: "red",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(
    <Label {...props}>test label</Label>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render label", () => {
  render(<Label {...props}>test label</Label>);
  const labelElement = screen.getByTestId("test-label-id");
  expect(labelElement).toBeInTheDocument();
});

test("should render label with text", () => {
  render(<Label {...props}>test label</Label>);
  const labelElement = screen.getByTestId("test-label-id");
  expect(labelElement).toBeInTheDocument();
  expect(labelElement).toHaveTextContent("test label");
});

test("should render label with modified style through props", () => {
  render(<Label {...modifiedProps} />);
  const labelElement = screen.getByTestId("test-label-id");
  expect(labelElement).toBeInTheDocument();
  expect(labelElement).toHaveStyle("color: red");
});
