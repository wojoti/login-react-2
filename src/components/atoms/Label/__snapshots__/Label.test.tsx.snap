// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<span
  className="sc-bcXHqe hySEci"
  data-testid="test-label-id"
>
  test label
</span>
`;
